### ISSUE TITLE
a11y: selected event popup hard to use with a keyboard

### ISSUE THREAD

**Issue by eatyourgreens** [OxfordRSE/gutenberg#408]
https://github.com/user-attachments/assets/c247ed48-c240-4b44-9340-5e519d6fc0bc

There's a few WCAG failures in the screen recording here, where I'm trying to use the toggle button (top left) from the keyboard.

1. The `ol` list uses invalid HTML (missing a required `li` child), so the toggle button is missed when a screen reader announces the contents of the list. ([WCAG SC 1.3.1 Info & relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html) level A.)
1. The button isn't using ARIA to tell screen readers that it will open a popup. After I press it, there's no announcement in a screen reader to tell me what's changed in the page state. ([WCAG SC 1.3.1 Info & relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html) level A.)
1. After I tab to the button, and open the popup from the keyboard, focus stays on the button so that I have to tab through the top navigation in order to get to the popup. In a screen  reader, I have no idea that a popup opened, unless I stumble upon it by accident at the end of the navigation menu. ([WCAG SC 2.4.3 Focus Order](https://www.w3.org/WAI/WCAG22/Understanding/focus-order.html) level A.)
1. After I've navigated into the popup, it doesn't have the `dialog` role, so isn't announced as a popup by screen readers, and can't be dismissed by pressing Esc. ([WCAG SC 1.3.1 Info & relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html) level A.)
1. The popup isn't modal, which is fine, but does mean I can tab out of it and continue to interact with the underlying page. This isn't a problem in a screen reader. However, a sighted keyboard user can tab to interactive elements in the event list, which are hidden behind the popup. ([WCAG SC 2.4.11 Focus not obscured](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html) level AA.)

At the moment, the popup fails to conform at level A, which means that it can't be used by students with disabilities.
